components/root/root.jsx

```diff
@@ -189,6 +189,14 @@ export default class Root extends React.Component {
             }
 
             this.setState({configLoaded: true});
+
+            const enableDeveloper = getConfig(store.getState()).EnableDeveloper;
+
+            if (enableDeveloper === 'true') {
```

Comment:
I'm wondering if we can pull this up "higher": as it stands now, any attempt to use `.length` on these types before the root component finishes loading won't trigger an error: that could represent a lot of uncovered code.

What if we move the definitions into a top-level file that's imported early, but with implementations that check (memoized?) for dev mode and thrown the error there? That has the advantage of not tying this to the root component at all.

Reply:
I don't know if that's possible since we need to wait until we have the client config to check whether or not developer mode is enabled. The root component is the one in charge of loading the config, so it seems like it's the ideal spot to put this. The only thing that we might be able to do that's further up is move this to be part of the `loadMeAndConfig` action, but that would only be earlier by a negligible amount